### PR TITLE
Add support for Seaborn figures

### PR DESCRIPTION
**Pull request by szysad**
## Before submitting checklist

- [x] Did you **update the CHANGELOG**? (not for test updates, internal changes/refactors or CI/CD setup)
- [x] Did you **ask the docs owner** to review all the user-facing changes?

### CODE REVIEW

**Review by normandy7 — DISMISSED** [review]

File: tests/e2e/plot_utils.py [hunk 1]
```diff
@@ -122,3 +123,11 @@ def generate_plotly_figure():
     plotly_fig = px.histogram(df, x="total_bill", y="tip", color="sex", marginal="rug", hover_data=df.columns)
 
     return plotly_fig
+
+
+def generate_seaborn_figure():
+    sample_size = 30
+    x = np.random.rand(sample_size) * 2 * np.pi
+    data = {"x": x, "y": np.sin(x), "c": np.random.randint(0, 2, sample_size), "arch": x > np.pi}
+    seaborn_fig = sns.relplot(data, x="x", y="y", hue="c", col="arch")
```
Comment (AleksanderWWW) [inline tests/e2e/plot_utils.py:132]: This doesn't seem to work with `seaborn` lower than `0.12.0`. 
I get `ValueError: Could not interpret value 'y' for parameter 'y'` for `0.11.2`
Could you check this and if so, fix the seaborn version in `pyproject.toml` to `>=0.12.0` ?

**Comment by AleksanderWWW** [comment]
@szysad 
My bad, forgot to let you know to start your branch name with your initials - let's do that in the future PRs :)

**Comment by AleksanderWWW** [comment]
### E2E :white_check_mark: 
https://github.com/neptune-ai/neptune-client/actions/runs/7389442981

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,7 @@
-## [UNRELEASED] 1.8.7
+## [UNRELEASED] 1.9.0
```
Comment (AleksanderWWW) [inline CHANGELOG.md:1]: Yes, in `1.8.6` it was also forgotten, but let's put it here as it is the convention throughout the entire changelog 
```suggestion
## [UNRELEASED] neptune 1.9.0
```

File: tests/unit/neptune/new/internal/utils/test_images.py [hunk 3]
```diff
@@ -148,6 +149,11 @@ def test_get_image_content_from_tensorflow_tensor(self):
         # expect
         self.assertEqual(get_image_content(image_tensor), self._encode_pil_image(expected_image))
 
+    def test_get_image_content_from_seaborn_figure(self):
```
[the remark this answers is not in the document]
Comment (AleksanderWWW) [reply tests/unit/neptune/new/internal/utils/test_images.py:152]: Same in the test for `html` export

File: src/neptune/internal/utils/images.py [hunk 4]
```diff
@@ -45,6 +45,13 @@
 from neptune.internal.utils.logger import logger
 
 _logger = logging.getLogger(__name__)
+SEABORN_GRID_CLASSES = set(
+    [
+        "FacetGrid",
+        "PairGrid",
+        "JointGrid",
+    ]
+)
```
Comment (AleksanderWWW) [inline src/neptune/internal/utils/images.py:54]: While this is technically correct, the syntax is slightly confusing and IDEs like PyCharm complain about it.
Let's replace it with a set literal:
`SEABORN_GRID_CLASSES = {"FacetGrid", "PairGrid", "JointGrid"}`

Also, what happened to 'pairplot' and 'jointplot'?

Comment (szysad) [reply src/neptune/internal/utils/images.py]: 'pairplot' and 'jointplot' are functions provided to user in [seaborn axisgrid](https://github.com/mwaskom/seaborn/blob/master/seaborn/axisgrid.py) but they always return instances of FacetGrid", "PairGrid", "JointGrid"

**Review by AleksanderWWW — APPROVED** [review]

File: pyproject.toml [hunk 5]
```diff
@@ -45,6 +45,7 @@ altair = { version = "*", optional = true }
 bokeh = { version = "*", optional = true }
 freezegun = { version = "*", optional = true }
 matplotlib = { version = "*", optional = true }
+seaborn = { version = "*", optional = true }
```
Comment (Raalsky) [inline pyproject.toml:48]: We have to add it as a part of `dev` group as well (see lines between 112 and 143) :wink:. This works as seaborn is probably a dependency of some of the other packages.

**Pull request merged**